docs/src/tutorial.md

```diff
@@ -1 +1,141 @@
+In addition, a `PartiallySeparableNLPModel` store the value of each element gradient to perform partitioned quasi-Newton afterward. 
+
+```@example PSNLP
+gx == NLPModels.grad(model, x)
+```
+
+The same procedure can be applied to `MathOptNLPModel`s:
+```@example PSNLP
+using JuMP, MathOptInterface, NLPModelsJuMP
+
+function jump_example(n::Int)
+  m = Model()
+  @variable(m, x[1:n])
+  @NLobjective(m, Min, sum((x[i] + x[i+1])^2 for i = 1:n-1))
+  evaluator = JuMP.NLPEvaluator(m)
+  MathOptInterface.initialize(evaluator, [:ExprGraph])
+  variables = JuMP.all_variables(m)
+  x0 = ones(n)
+  JuMP.set_start_value.(variables, x0)
+  nlp = MathOptNLPModel(m)
+  return nlp
+end
+
+jumpnlp_example = jump_example(n)
+pqn_jumpnlp = PartiallySeparableNLPModel(jumpnlp_example)
+
+fx = NLPModels.obj(pqn_jumpnlp, x) # compute the obective function
+gx = NLPModels.grad(pqn_jumpnlp, x) # compute the gradient
+```
+
+## A partitioned quasi-Newton `NLPModel`
+By defining a `PartiallySeparableNLPModel` you allocate a partitioned quasi-Newton update, which is implemented in `PartitionedStructures.jl`.
+
+The Hessian approximation of each element function $\hat{f}_i (y) = (y_1 + y_2)^2$ is initially set to an identity matrix. 
+The contribution of every element Hessian approximation is accumulated as
+```math
+\left [
+\begin{array}{ccc}
+  \left ( \begin{array}{cc}
+    1 & \\
+    & 1 \\ 
+  \end{array} \right ) & & \\
+  & 0 & \\
+  & & 0 \\
+\end{array}
+\right ] 
++ 
+\left [
+\begin{array}{ccc}
+  0 & & \\
+  & \left ( \begin{array}{cc}
+    1 & \\
+    & 1 \\ 
+  \end{array} \right ) & \\
+  & & 0 \\
+\end{array}
+\right ]
++ 
+\left [
+\begin{array}{ccc}
+  0 & & \\
+  & 0 & \\
+  & & \left ( \begin{array}{cc}
+    1 & \\
+    & 1 \\ 
+  \end{array} \right )\\
+\end{array}
+\right ]
+```
+The accumulated matrix can be visualized with:
+```@example PSNLP
+Matrix(hess_approx(pqn_jumpnlp))
+```
+
+You can specify the partitioned quasi-Newton update with the optional argument `name`:
+```julia
+PartiallySeparableNLPModel(jumpnlp_example; name=:plse) # by default
+```
+The possible variants are: `:pbfgs, :psr1, :pse, :plbfgs, :plsr1` and `:plse`, see the [PartitionedStructures.jl tutorial](https://juliasmoothoptimizers.github.io/PartitionedStructures.jl/dev/tutorial/) for more details.
+
+Then, you can update the partitioned quasi-Newton approximation with the pair `x,s`:
+```@example PSNLP
+s = rand(n)
+update_nlp(pqn_adnlp, x, s)
+```
+and you can perform the partitioned matrix-vector product with:
+```@example PSNLP
+v = ones(n)
+Bv = hprod(pqn_adnlp, x, v)
+```
+
+An in-place variant helps define a `LinearOperator` (see [LinearOperators](https://github.com/JuliaSmoothOptimizers/LinearOperators.jl)) from a `PartiallySeparableNLPModel`:
+```@example PSNLP
+using LinearOperators
+T = eltype(x)
+B = LinearOperator(T, n, n, true, true, ((Hv, v) -> hprod!(pqn_adnlp, x, v, Hv)))
+B*v
```

Comment:
This should not be necessary. Normally, `hess_op` should "just work" if your `hprod!()` follows the NLPModels API.